src/machine.rs

```diff
@@ -24,10 +24,22 @@ use hyperlocal::{UnixClientExt, UnixConnector, Uri};
 #[derive(Debug)]
 pub struct Machine<'m> {
     config: Config<'m>,
-    pid: Option<i32>,
+    state: MachineState,
     client: Client<UnixConnector>,
 }
 
+/// VM state
+#[derive(Debug, Clone, Copy)]
+pub enum MachineState {
+    /// Machine is not started or already shut down
+    Off,
+    /// Machine is running
+    On {
```

Comment:
I know I recommended the names so my bad but let's use the same names as [libvirt](https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainState) does: `SHUTOFF` and `RUNNING`. If we do add other variants here, I think `Off` and `On` might become confusing.